File: client/scalaz/src/main/scala/org/openapitools/client/api/RecordList.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import RecordList._

case class RecordList (
  metadata: Option[Metadata],
pagination: Option[Pagination],
data: Option[List[Record]])

object RecordList {
  import DateTimeCodecs._

  implicit val RecordListCodecJson: CodecJson[RecordList] = CodecJson.derive[RecordList]
  implicit val RecordListDecoder: EntityDecoder[RecordList] = jsonOf[RecordList]
  implicit val RecordListEncoder: EntityEncoder[RecordList] = jsonEncoderOf[RecordList]
}

File: API/Client/RecordList.js
goog.provide('API.Client.RecordList');

/**
 * @record
 */
API.Client.RecordList = function() {}

/**
 * @type {!API.Client.Metadata}
 * @export
 */
API.Client.RecordList.prototype.metadata;

/**
 * @type {!API.Client.Pagination}
 * @export
 */
API.Client.RecordList.prototype.pagination;

/**
 * @type {!Array<!API.Client.Record>}
 * @export
 */
API.Client.RecordList.prototype.data;

